src/ngraph/runtime/cache.hpp

```diff
@@ -0,0 +1,61 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <list>
+#include <mutex>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include "ngraph/function.hpp"
+#include "ngraph/runtime/executable.hpp"
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        class LRUCache : public std::enable_shared_from_this<LRUCache>
```

Comment:
This needs a doc string describing what it is caching. Just calling the file cache.?pp is not helpful. What is it caching? Change the name and document better.
Calling the cache LRUCache is not a good idea. You should not expose the implementation method in the name of the cache. What if you change to a new algorithm, no LRU? Do you rename the class breaking users or do you leave it misleadingly named LRUCache. The name of the class should be WHAT it is caching, not HOW it is caching it.